### ISSUE TITLE
No longer works

### ISSUE THREAD

**Issue by geeclectic** [crot4lus/project-dropdown#9]
setActiveProject no longer targets the project-root and it seems to empty the ol list within all other project-root elements.

**Comment by GlennTaylorDigital** [comment]
Hi, which version are you running? I can't seem to replicate this on either of my computers or on a colleagues computer.

Cheers

**Comment by geeclectic** [comment]
It is on-going though many updates and is still an issue today. Currently I am using Atom 1.12.7, Project Dropdown 0.8.0. I have disabled or uninstalled all but core packages and the problem persists as described.

In the debugger, there is a line in 'setActiveProject' that returns null that should be setting the target project-root to visible:

$(".tree-view-scroller .directory.project-root .header.list-item span.name[data-path='" + path + "']").parents('li').show();

I have tried breaking this down into steps, and all steps return null.

I made a little progress porting this method to native JS, but results were mixed (it would show the active project, then almost immediately hide it again).

Perhaps this can help you diagnose.

**Comment by GlennTaylorDigital** [comment]
Can you try opening your debugger and clicking File -> Add Project and add a directory and see if the issue / error re-occurs? If so can you provide me a screenshot?

Are you running Mac or Windows? All devices I have in our offices that run it are on Mac and they all seem to be OK. I guess difference in setup could cause issues but if you've disabled all other extensions it *should* be fine.

**Comment by geeclectic** [comment]
I use PC elusively. My main PC is a new build too. However, this is not limited to one PC, but all (3 here at home and half a dozen at the office, all with mixed setups). I work on dozens of projects at any given time, so am adding and removing projects constantly, however it fails if I have 1 or 30 projects loaded, I even loaded atom fresh with no projects on my media PC and loaded one project, and the problem persists.

I just added a another project folder as requested though, and all came up blank, with nothing in the console:

![capture1](https://cloud.githubusercontent.com/assets/9344547/21384910/4745dba2-c7d1-11e6-9fbc-b928af25941f.PNG)

As you can see (as mentioned in my first comment above), each project root still has 'display:none', even on the current 'expanded' li node. Any project I select receives the 'expanded' class, but remains 'display:none' in the style attribute:

![capture2](https://cloud.githubusercontent.com/assets/9344547/21384959/78e7ffaa-c7d1-11e6-85e5-a1b1d2223558.PNG)

I hope this helps. It would be great to have this feature back!

**Comment by GlennTaylorDigital** [comment]
Apologies for the delay - I've just invested in a Surface Pro 4 so used that to check and debug the issue.

Turns out that the incompatibility with Windows is caused by the path structure, on Mac the project path would be something like /Users/user/Projects/project-name - and as you will be aware forward slashes are acceptable in strings, however Windows paths are structured like c:\users\username\Documents - the backslash is obviously being treated as an escaping character which throws out the jQuery selector you mentioned above.

I've tweaked the code so it searches and replaces \ with \\ so it escapes the backslash.

Check out the latest version and let me know if you have any other issues.
